casanovo/denovo/dataloaders.py

```diff
@@ -146,68 +278,37 @@ def _make_loader(
+            # precision=torch.float32,
+            pin_memory=True,
         )
 
     def train_dataloader(self) -> torch.utils.data.DataLoader:
         """Get the training DataLoader."""
-        return self._make_loader(
-            self.train_dataset, self.train_batch_size, shuffle=True
-        )
+        return self._make_loader(self.train_dataset, self.shuffle)
 
     def val_dataloader(self) -> torch.utils.data.DataLoader:
         """Get the validation DataLoader."""
-        return self._make_loader(self.valid_dataset, self.eval_batch_size)
+        return self._make_loader(self.valid_dataset)
 
     def test_dataloader(self) -> torch.utils.data.DataLoader:
         """Get the test DataLoader."""
-        return self._make_loader(self.test_dataset, self.eval_batch_size)
+        return self._make_loader(self.test_dataset)
 
     def predict_dataloader(self) -> torch.utils.data.DataLoader:
         """Get the predict DataLoader."""
-        return self._make_loader(self.test_dataset, self.eval_batch_size)
+        return self._make_loader(self.test_dataset)
 
 
-def prepare_batch(
-    batch: List[Tuple[torch.Tensor, float, int, str]]
-) -> Tuple[torch.Tensor, torch.Tensor, np.ndarray]:
+def scale_to_unit_norm(spectrum):
     """
-    Collate MS/MS spectra into a batch.
-
-    The MS/MS spectra will be padded so that they fit nicely as a tensor.
-    However, the padded elements are ignored during the subsequent steps.
-
-    Parameters
-    ----------
-    batch : List[Tuple[torch.Tensor, float, int, str]]
-        A batch of data from an AnnotatedSpectrumDataset, consisting of for each
-        spectrum (i) a tensor with the m/z and intensity peak values, (ii), the
-        precursor m/z, (iii) the precursor charge, (iv) the spectrum identifier.
-
-    Returns
-    -------
-    spectra : torch.Tensor of shape (batch_size, n_peaks, 2)
-        The padded mass spectra tensor with the m/z and intensity peak values
-        for each spectrum.
-    precursors : torch.Tensor of shape (batch_size, 3)
-        A tensor with the precursor neutral mass, precursor charge, and
-        precursor m/z.
-    spectrum_ids : np.ndarray
-        The spectrum identifiers (during de novo sequencing) or peptide
-        sequences (during training).
+    Scaling function used in Casanovo
+    slightly differing from the depthcharge implementation
     """
-    spectra, precursor_mzs, precursor_charges, spectrum_ids = list(zip(*batch))
-    spectra = torch.nn.utils.rnn.pad_sequence(spectra, batch_first=True)
-    precursor_mzs = torch.tensor(precursor_mzs)
-    precursor_charges = torch.tensor(precursor_charges)
-    precursor_masses = (precursor_mzs - 1.007276) * precursor_charges
-    precursors = torch.vstack(
-        [precursor_masses, precursor_charges, precursor_mzs]
-    ).T.float()
-    return spectra, precursors, np.asarray(spectrum_ids)
+    spectrum._inner._intensity = spectrum.intensity / np.linalg.norm(
+        spectrum.intensity
+    )
```

Comment:
Yes, we were also thinking the parentheses are misplaced in Depthcharge. If you change that version, I would directly import the function from Depthcharge and leave it as it is otherwise.